packages/swing-store/src/snapStore.js

```diff
@@ -250,20 +255,28 @@ export function makeSnapStore(
             compressedSize,
             compressedSnapshot,
           );
-          const rec = snapshotRec(vatID, snapPos, hash, 1);
-          const exportKey = snapshotMetadataKey(rec);
           noteExport(exportKey, JSON.stringify(rec));
           noteExport(
             currentSnapshotMetadataKey(rec),
             snapshotArtifactName(rec),
           );
         });
 
+        let archiveWriteSeconds;
+        if (archiveSnapshot) {
+          ({ duration: archiveWriteSeconds } = await measureSeconds(
+            async () => {
+              await archiveSnapshot(exportKey, compressedSnapshot);
+            },
+          ));
+        }
```

Comment:
I was originally thinking we'd want to create a `PassThrough` stream, pipe the `gzip` stream above to it just before passing `gzip` to `buffer` (aka tee the gzip stream into the buffer and the archiver). That would avoid delaying the overall save operation with a sequential step.

Reply:
I didn't want it to muddy our compressSeconds metric, which currently does not include any writes to disk. And we run sqlSaveSnapshot with the full compressed contents anyway, so it didn't seem worthwhile to optimize one half of the tee while leaving the other fundamentally non-optimized. But that intuition might be wrong, and I look forward to archiveWriteSeconds data upon which we can make a more informed decision.